File: frontend/iosApp/Domains/Sources/Domains/MimeType.swift
import Foundation

public enum MimeType: String, Sendable {
    case jpeg = "image/jpeg"

    public var fileExtension: String {
        switch self {
        case .jpeg: return "jpg"
        }
    }

    public func fileName(for id: UUID) -> String {
        "\(id).\(fileExtension)"
    }
}

File: frontend/shared/src/commonMain/kotlin/com/example/memoriesapp/domain/MimeType.kt
package com.example.memoriesapp.domain

import com.example.memoriesapp.core.LocalId

enum class MimeType(val value: String) {
    JPEG("image/jpeg");

    val fileExtension: String
        get() = when (this) {
            JPEG -> "jpg"
        }

    fun fileName(id: LocalId): String = "${id}.${fileExtension}"
}
